File: backend/schemas/book.py
from pydantic import BaseModel
from schemas.genre import GenreOut


class BookCreate(BaseModel):
    title: str
    author: str
    year: int | None = None
    genre_id: int | None = None


class BookOut(BaseModel):
    id: int
    title: str
    author: str
    year: int | None
    genre: GenreOut | None

    model_config = {"from_attributes": True}


class BookDetail(BookOut):
    avg_rating: float | None = None
    rating_count: int = 0

File: frontend/src/types/book.ts
export interface Genre {
  id: number
  name: string
}

export interface Book {
  id: number
  title: string
  author: string
  year: number | null
  genre: Genre | null
}

export interface BookDetail extends Book {
  avg_rating: number | null
  rating_count: number
  tags: { id: number; name: string }[]
}

export interface BookCreate {
  title: string
  author: string
  year?: number
  genre_id?: number
}
